File: python/graphical/app/prime_factors.py
#!/usr/bin/env python3
from math import sqrt

class prime_factors(object):
    def __init__(self):
        self.factors = []

    def decompose(self, n):
        while(n%2 == 0):
            self.factors.append(2)
            n /= 2

        for i in range(3,int(sqrt(n))+1):
            while(n%i == 0):
                self.factors.append(i)
                n /= i

        if(n > 2):
            self.factors.append(int(n))

def main():
    test = prime_factors()
    test.decompose(10124215)
    print(test.factors)

if __name__=='__main__':
    main()

File: cpp/graphical/prime_decomposer/prime_factors.cpp
#include "prime_factors.h"
#include <stdio.h>
#include <math.h>
#include <vector>
using namespace std;

prime_factors::prime_factors(){

}

void prime_factors::decompose(long n){
    while(n%2 == 0){
        factors.push_back(2);
        n = n/2;
    }

    for(int i=3; i<sqrt(n); i++){
        while(n%i == 0){
            factors.push_back(i);
            n = n/i;
        }
    }

    if(n>2){
        factors.push_back(n);
    }
}

vector<int> prime_factors::get_factors(){
    return factors;
}
